include/devicetree/reset.h

```diff
@@ -0,0 +1,246 @@
+/**
+ * @file
+ * @brief Reset Controller Devicetree macro public API header file.
+ */
+
+/*
+ * Copyright (c) 2022, Andrei-Edward Popa
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_DEVICETREE_RESET_H_
+#define ZEPHYR_INCLUDE_DEVICETREE_RESET_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup devicetree-reset-controller Devicetree Reset Controller API
+ * @ingroup devicetree
+ * @{
+ */
+
+/**
+ * @brief Get the node identifier for the controller phandle from a
+ *        "resets" phandle-array property at an index
+ *
+ * Example devicetree fragment:
+ *
+ *     reset1: reset-controller@... { ... };
+ *
+ *     reset2: reset-controller@... { ... };
+ *
+ *     n: node {
+ *             resets = <&reset1 10>, <&reset2 20>;
+ *     };
+ *
+ * Example usage:
+ *
+ *     DT_RESET_CTLR_BY_IDX(DT_NODELABEL(n), 0)) // DT_NODELABEL(reset1)
+ *     DT_RESET_CTLR_BY_IDX(DT_NODELABEL(n), 1)) // DT_NODELABEL(reset2)
+ *
+ * @param node_id node identifier
+ * @param idx logical index into "resets"
+ * @return the node identifier for the reset controller referenced at
+ *         index "idx"
+ * @see DT_PHANDLE_BY_IDX()
+ */
+#define DT_RESET_CTLR_BY_IDX(node_id, idx) \
+	DT_PHANDLE_BY_IDX(node_id, resets, idx)
```

Comment:
Usually we do not pollute the `DT_*` namespace, so you want to rename these functions something like `RESET_DT_...` instead.

Reply:
So DT_RESET_CTLR_BY_IDX need to be remaned as RESET_CTRL_DT_BY_IDX?

Reply:
`RESET_DT_CTLR_BY_IDX`